tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_assign.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import pytest
+import tt_lib
+from tests.tt_eager.python_api_testing.sweep_tests import (
+    comparison_funcs,
+)
+from loguru import logger
+
+
+@pytest.mark.parametrize(
+    "input_shapes",
+    (
+        (torch.Size([1, 1, 32, 32])),
+        (torch.Size([1, 1, 320, 384])),
+        (torch.Size([1, 3, 320, 384])),
+    ),
+)
+def test_bw_binary_assign(input_shapes, device):
+    torch.manual_seed(0)
+    in_data = torch.randn(input_shapes, requires_grad=True).bfloat16()
+    other_data = torch.randn(input_shapes, requires_grad=True).bfloat16()
+    grad_data = torch.randn(input_shapes).bfloat16()
+
+    grad_tensor = (
+        tt_lib.tensor.Tensor(grad_data, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
+    )
+
+    input_tensor = (
+        tt_lib.tensor.Tensor(in_data, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
+    )
+
+    other_tensor = (
+        tt_lib.tensor.Tensor(other_data, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
+    )
+
+    tt_output_tensor_on_device = tt_lib.tensor.binary_assign_bw(grad_tensor, input_tensor, other_tensor)
+    tt_output_tensor = tt_output_tensor_on_device[0].cpu().to(tt_lib.tensor.Layout.ROW_MAJOR).to_torch()
+
+    in_data.retain_grad()
+
+    pyt_y = torch.clone(in_data)
+
+    pyt_y.backward(gradient=grad_data)
+
+    golden_output_tensor = in_data.grad
+
+    comp_pass, _ = comparison_funcs.comp_equal(golden_output_tensor, tt_output_tensor)
+    _, comp_out = comparison_funcs.comp_allclose_and_pcc(golden_output_tensor, tt_output_tensor)
```

Comment:
comp_equal returns the atol, rtol, pcc in it's output, so you dont need to run comp_allclose_and_pcc separately to get comp_out

Reply:
Done